docs/rfcs/023-the-state-of-pageserver-tenant-relocation.md

```diff
@@ -0,0 +1,230 @@
+# The state of pageserver tenant relocation
+
+Created on 17.03.23
+
+## Motivation
+
+There were previous write ups on the subject. The design of tenant relocation was planned at the time when we had quite different landscape. I e there was no on-demand download/eviction. They were on the horizon but we still planned for cases when they were not available. Some other things have changed. Now safekeepers offload wal to s3 so we're not risking overflowing their disks. Having all of the above, it makes sense to recap and take a look at the options we have now, which adjustments we'd like to make to original process, etc.
+
+Related (in chronological order):
+
+- Tracking issue with initial discussion: [#886](https://github.com/neondatabase/neon/issues/886)
+- [015. Storage Messaging](015-storage-messaging.md)
+- [020. Pageserver S3 Coordination](020-pageserver-s3-coordination.md)
+
+## Summary
+
+The RFC consists of a walkthrough of prior art on tenant relocation and corresponding problems. It describes 3 approaches.
+
+1. Simplistic approach that uses ignore and is the fastest to implement. The main downside is a requirement of short downtime.
+2. More complicated approach that avoids even short downtime.
```

Comment:
I see "Simplistic approach" that I'd map to `1.`, see the "Lease based approach" that I'd map to `3.`, where is `2.` in the RFC?

Reply:
The order is not chronological, maybe I should change that. Rfc starts with discussing 2 then goes to compromise 1 and then discusses broader solution 3

Reply:
I'll put a note that order in text is different

Reply:
Done